packages/core/types/index.js

```diff
@@ -299,6 +302,42 @@ export interface Asset extends BaseAsset {
   +stats: Stats;
 }
 
+// TODO: figure out naming
+export interface IConfig {
+  searchPath: FilePath;
+  env: Environment;
+  options: ParcelOptions;
+  resolvedPath: ?FilePath;
+  // $FlowFixMe could be anything
+  result: ?any;
+  resultHash: ?string;
+  includedFiles: Set<FilePath>;
+  watchGlob: ?Glob;
+  devDeps: Map<PackageName, ?string>;
+  pkg: ?PackageJSON;
+
+  setResolvedPath(filePath: FilePath): void;
+  // $FlowFixMe could be anything
+  setResult(result: any): void; // TODO: fix
+  setResultHash(resultHash: string): void;
+  addIncludedFile(filePath: FilePath): void;
+  setDevDep(name: PackageName, version?: Semver): void;
+  setWatchGlob(glob: string): void;
+  getConfigFrom(
+    searchPath: FilePath,
+    filePaths: Array<FilePath>,
+    options: ?{packageKey?: string, parse?: boolean}
+  ): Promise<ThirdPartyConfig | null>;
+  getConfig(
+    filePaths: Array<FilePath>,
+    options: ?{packageKey?: string, parse?: boolean}
+  ): Promise<ThirdPartyConfig | null>;
+  getPackage(): Promise<PackageJSON | null>;
+  isSource(): Promise<boolean>;
+  shouldRehydrate(): void;
+  shouldReload(): void;
```

Comment:
maybe a getter/setter for these instead of methods?

Reply:
Those are the functions that the plugin uses to tell core that the config should rehydrated or reloaded. The functions just set a variable on the class that core then reads later. That's why the variables weren't named in a way that makes it clear that they are booleans. I wasn't sure what else to name them that wouldn't conflict with the method names.

Reply:
Yeah I was thinking we could have both `get shouldReload()` and `set shouldReload()`. The internal variable would become private.